content/data-loading.md

```diff
@@ -0,0 +1,556 @@
+
+However, we can write publications that are properly reactive to changes across collections. To do this, we use the [`reywood:publish-composite`](https://atmospherejs.com/reywood/publish-composite) package. 
+
+The way this package works is to first establish a cursor on one collection, and then explicitly set up a second level of cursors on a second collection with the results of the first cursor.
+
+XXX: this example isn't actually in the app yet: https://github.com/meteor/todos/issues/48
+
+```js
+Meteor.publishComposite('list/todos', function(listId) {
+  check(listId, String);
+
+  const userId = this.userId;
+  return {
+    find() {
+      const query = {
+        _id: listId,
+        $or: [{userId: {$exists: false}}, {userId}]
+      };
+
+      return Lists.find(query);
+    },
+    children: [{
+      find(list) {
+        return Todos.find({listId: list._id});
+      }
+    }]
+  };
+});
+```
+
+In this example, we write a complicated query to make sure that we only ever find a list if we are allowed to see it, then, once per list we find (which can be one or zero times depending on access), we publish the todos for that list. Publish Composite takes care of stopping and starting the dependent cursors if the list stops matching the original query or otherwise.
+
+## Complex Authorization in publications
+
+We can also use `publish-composite` to perform complex authorization in publications. For instance, consider if we had a `admin/list/todos` publication that allowed an admin to bypass default publication's security for users with an `admin` flag set.
+
+We might want to write:
+
+```js
+Meteor.publish('admin/list/todos', function(listId) {
+  check(listId, String);
+
+  const user = Meteor.users.findOne(this.userId);
+
+  if (user && user.admin) {
+    return [
+      Lists.find(listId),
+      Todos.find({listId});
+    ];
+  } else {
+    return this.ready();
+  }
+});
+```
+
+However, due to the same reasons discussed above, the publication *will not re-run* if the user's `admin` status changes. If this is something that is likely to happen and reactive changes are needed, then we'll need to make the publication reactive. We can do this via the same technique as above however:
+
+```js
+Meteor.publishComposite('admin/list/todos', function(listId) {
+  check(listId, String);
+
+  const userId = this.userId;
+  return {
+    find() {
+      return Meteor.users.find({userId, admin: true});
+    },
+    children: [{
+      find() {
+        return [
+          Lists.find(listId),
+          Todos.find({listId});
+        ];
+      }  
+    }]
+  };
+});
+```
+
+## The low level publish API
```

Comment:
I'd phrase this in a use-case way - "Writing custom publications with the low-level publication API" or something?